aws/src/test/java/org/apache/iceberg/aws/TestAwsProperties.java

```diff
@@ -123,4 +129,74 @@ public void testS3FileIoDeleteBatchSizeTooSmall() {
         "Deletion batch size must be between 1 and 1000",
         () -> new AwsProperties(map));
   }
+
+  @Test
+  public void testS3FileIoCredentialsConfiguration() {
+    Map<String, String> properties = Maps.newHashMap();
+    ArgumentCaptor<AwsCredentialsProvider> awsCredentialsProviderArgumentCaptor =
+        ArgumentCaptor.forClass(AwsCredentialsProvider.class);
+
+    // set nothing
+    AwsProperties testAwsPropertiesAllNull = new AwsProperties(properties);
+    S3ClientBuilder mockS3ClientBuilderAllNull = Mockito.mock(S3ClientBuilder.class);
+    testAwsPropertiesAllNull.applyS3CredentialConfigurations(mockS3ClientBuilderAllNull);
+    Mockito.verify(mockS3ClientBuilderAllNull)
+        .credentialsProvider(awsCredentialsProviderArgumentCaptor.capture());
+    AwsCredentialsProvider capturedAwsCredentialsProviderAllNull =
+        awsCredentialsProviderArgumentCaptor.getValue();
+    Assert.assertTrue(
+        "Should use default credentials if nothing is set",
+        capturedAwsCredentialsProviderAllNull instanceof DefaultCredentialsProvider);
+
+    // set access key id and secret access key
```

Comment:
looks like we can separate these to 3 different tests for different behaviors

Reply:
Thank you for your suggestion. I separated them to :`testS3FileIoDefaultCredentialsConfiguration`, `testS3FileIoBasicCredentialsConfiguration` and `testS3FileIoSessionCredentialsConfiguration`